packages/tpetra/tsqr/src/Tsqr_KokkosNodeTsqr.hpp

```diff
@@ -1199,85 +1112,16 @@ namespace TSQR {
-    ///   will be filled in with default values.
-    KokkosNodeTsqr (const Teuchos::RCP<node_type>& node,
-                    const Teuchos::RCP<Teuchos::ParameterList>& params) :
-      node_ (node)
-    {
-      setParameterList (params);
-    }
-
-    /// \brief Constructor (with user-specified parameters but no node).
-    ///
-    /// This version of the constructor sets the Kokkos Node instance
-    /// to null.  You must call \c setNode() with a valid Kokkos Node
-    /// instance before you can invoke any methods that perform
-    /// computations.
-    ///
     /// \param params [in/out] List of parameters.  Missing parameters
     ///   will be filled in with default values.
-    KokkosNodeTsqr (const Teuchos::RCP<Teuchos::ParameterList>& params) :
-      node_ (Teuchos::null)
+    KokkosNodeTsqr (const Teuchos::RCP<Teuchos::ParameterList>& params = Teuchos::null)
     {
       setParameterList (params);
     }
 
-    /// \brief Constructor (sets default parameters).
-    ///
-    /// \param node [in] Kokkos Node instance.  If you don't have this
-    ///   yet, you can set it to null and call \c setNode() later once
-    ///   you have the Node instance.
-    KokkosNodeTsqr (const Teuchos::RCP<node_type>& node) :
-      node_ (node)
-    {
-      setParameterList (Teuchos::null);
-    }
-
-    /// \brief Default constructor (sets default parameters).
-    ///
-    /// This version of the constructor sets the Kokkos Node instance
-    /// to null.  You must call \c setNode() with a valid Kokkos Node
-    /// instance before you can invoke any methods that perform
-    /// computations.
-    KokkosNodeTsqr () : node_ (Teuchos::null)
-    {
-      setParameterList (Teuchos::null);
-    }
-
-    /// \brief Set the Kokkos Node instance.
-    ///
-    /// You can't compute anything until you set the Kokkos Node
-    /// instance.
-    ///
-    /// \note The whole reason for allowing initialization of the
-    ///   Kokkos Node instance after construction is so that this
-    ///   class can implement \c Teuchos::ParameterListAcceptor.
-    ///   ParameterListAcceptor's getValidParameters() is an instance
-    ///   method, not a class method, so the object has to be
-    ///   instantiated before getValidParameters() can be called.  \c
-    ///   NodeTsqrFactory in turn needs to call getValidParameters()
-    ///   so that callers can get a default parameter list before
-    ///   instantiating the NodeTsqr subclass instance.  However,
-    ///   NodeTsqrFactory doesn't have the Kokkos Node instance until
-    ///   TSQR gets a multivector to factor.
-    void setNode (const Teuchos::RCP<node_type>& node) {
-      node_ = node;
-    }
-
     /// \brief Whether this object is ready to perform computations.
-    ///
-    /// It is <i>not</i> ready if the Kokkos Node instance has not yet
-    /// been set.
     bool ready() const {
-      return ! getNode().is_null();
+      return true;
```

Comment:
Why is this always true?

Reply:
@mperego The class no longer depends on a Kokkos 1.0 Node object.  It used to be that the class wasn't "ready" until somebody called it with a `nonnull RCP<Node>`.